drivers/display/display_st7701.c

```diff
@@ -0,0 +1,447 @@
+	st7701_generic_write(dev, cfg->gip_eb, sizeof(cfg->gip_eb));
+	st7701_generic_write(dev, cfg->gip_ec, sizeof(cfg->gip_ec));
+	st7701_generic_write(dev, cfg->gip_ed, sizeof(cfg->gip_ed));
+
+	/* Bank1 setting */
+	const uint8_t _FF2[] = {DSI_CMD2BKX_SEL, 0x77, 0x01, 0x00, 0x00, DSI_CMD2BKX_SEL_NONE};
+
+	st7701_generic_write(dev, _FF2, sizeof(_FF2));
+
+	/* Exit sleep mode */
+	ret = st7701_dcs_write(dev, MIPI_DCS_EXIT_SLEEP_MODE, NULL, 0);
+	if (ret < 0) {
+		return ret;
+	}
+
+	k_msleep(50);
+
+	/* Set pixel color format */
+	switch (data->dsi_pixel_format) {
+	case MIPI_DSI_PIXFMT_RGB565:
+		buf[0] = MIPI_DCS_PIXEL_FORMAT_16BIT;
+		break;
+	case MIPI_DSI_PIXFMT_RGB888:
+		buf[0] = MIPI_DCS_PIXEL_FORMAT_24BIT;
+		break;
+	default:
+		LOG_ERR("Unsupported pixel format 0x%x!", data->dsi_pixel_format);
+		return -ENOTSUP;
+	}
+
+	ret = st7701_dcs_write(dev, MIPI_DCS_SET_PIXEL_FORMAT, buf, 1);
+	if (ret < 0) {
+		return ret;
+	}
+
+	buf[0] = 0x00;
+	buf[1] = 0x00;
+	sys_put_be16(data->xres, (uint8_t *)&buf[2]);
+	ret = st7701_dcs_write(dev, MIPI_DCS_SET_COLUMN_ADDRESS, buf, 4);
+	if (ret < 0) {
+		return ret;
+	}
+
+	buf[0] = 0x00;
+	buf[1] = 0x00;
+	sys_put_be16(data->yres, (uint8_t *)&buf[2]);
+	ret = st7701_dcs_write(dev, MIPI_DCS_SET_PAGE_ADDRESS, buf, 4);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Backlight control */
+	buf[0] = ST7701_WRCTRLD_BCTRL | ST7701_WRCTRLD_DD | ST7701_WRCTRLD_BL;
+	ret = st7701_dcs_write(dev, MIPI_DCS_WRITE_CONTROL_DISPLAY, buf, 1);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Adaptive brightness control */
+	buf[0] = ST7701_WRCABC_UI;
+	ret = st7701_dcs_write(dev, MIPI_DCS_WRITE_POWER_SAVE, buf, 1);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Adaptive brightness control minimum brightness */
+	buf[0] = 0xFF;
+	ret = st7701_dcs_write(dev, MIPI_DCS_SET_CABC_MIN_BRIGHTNESS, buf, 1);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Brightness */
+	buf[0] = 0xFF;
+	ret = st7701_dcs_write(dev, MIPI_DCS_SET_DISPLAY_BRIGHTNESS, buf, 1);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Display On */
+	ret = st7701_dcs_write(dev, MIPI_DCS_SET_DISPLAY_ON, NULL, 0);
+	if (ret < 0) {
+		return ret;
+	}
+
+	return 0;
+}
+
+static int st7701_blanking_on(const struct device *dev)
+{
+	const struct st7701_config *cfg = dev->config;
+	int ret;
+
+	if (cfg->backlight.port != NULL) {
+		ret = gpio_pin_set_dt(&cfg->backlight, 0);
+		if (ret) {
+			LOG_ERR("Disable backlight failed! (%d)", ret);
+			return ret;
+		}
+	}
+
+	return st7701_dcs_write(dev, MIPI_DCS_SET_DISPLAY_OFF, NULL, 0);
+}
+
+static int st7701_blanking_off(const struct device *dev)
+{
+	const struct st7701_config *cfg = dev->config;
+	int ret;
+
+	if (cfg->backlight.port != NULL) {
+		ret = gpio_pin_set_dt(&cfg->backlight, 1);
+		if (ret) {
+			LOG_ERR("Enable backlight failed! (%d)", ret);
+			return ret;
+		}
+	}
+
+	return st7701_dcs_write(dev, MIPI_DCS_SET_DISPLAY_ON, NULL, 0);
+}
+
+static int st7701_write(const struct device *dev, uint16_t x, uint16_t y,
```

Comment:
No need for this function- the display API will return `-ENOTSUP` if you simply leave the API function pointer as `NULL` in the driver API struct

Reply:
Removed, thanks!